scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5158 @@
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-0.5, b=0.5)
+
+        Retrieve the support of the distribution:
+
+        >>> X.support()
+        (-0.5, 0.5)
+
+        For a distribution with infinite support,
+
+        >>> X = stats.Normal()
+        >>> X.support()
+        (-inf, inf)
+
+        Due to underflow, the numerical value returned by the PDF may be zero
+        even for arguments within the support, even if the true value is
+        nonzero. In such cases, the log-PDF may be useful.
+
+        >>> X.pdf([-100., 100.])
+        array([0., 0.])
+        >>> X.logpdf([-100., 100.])
+        array([-5000.91893853, -5000.91893853])
+
+        Use cases for the log-CDF and related methods are analogous.
+
+        """
+        # If this were a `cached_property`, we couldn't update the value
+        # when the distribution parameters change.
+        # Caching is important, though, because calls to _support take 1~2 µs
+        # even when `a` and `b` are already the same shape.
+        if self._support_cache is not None:
+            return self._support_cache
+
+        a, b = self._support(**self._parameters)
+        if a.shape != self._shape:
+            a = np.broadcast_to(a, self._shape)
+        if b.shape != self._shape:
+            b = np.broadcast_to(b, self._shape)
+
+        if self._any_invalid:
+            a, b = np.asarray(a).copy(), np.asarray(b).copy()
+            a[self._invalid], b[self._invalid] = np.nan, np.nan
+            a, b = a[()], b[()]
+
+        support = (a, b)
+
+        if self.cache_policy != _NO_CACHE:
+            self._support_cache = support
+
+        return support
+
+    def _support(self, **kwargs):
+        # Computes the support given distribution parameters
+        a, b = self._variable.domain.get_numerical_endpoints(kwargs)
+        if len(kwargs):
+            # the parameters should all be of the same dtype and shape at this point
+            vals = list(kwargs.values())
+            shape = vals[0].shape
+            a = np.broadcast_to(a, shape) if a.shape != shape else a
+            b = np.broadcast_to(b, shape) if b.shape != shape else b
+        return self._preserve_type(a), self._preserve_type(b)
+
+    @_set_invalid_nan_property
+    def logentropy(self, *, method=None):
+        r"""Logarithm of the differential entropy
+
+        In terms of probability density function :math:`f(x)` and its support
+        :math:`\chi`, the differential entropy of a random variable :math:`X` is:
+
+        .. math::
+
+            h(X) = - \int_{\chi} f(x) \log f(x) dx
+
+        `logentropy` computes the logarithm of the differential entropy,
+        :math:`log(h(X))`, but it may be numerically favorable compared to the
+        naive implementation (computing :math:`h(X)` and taking the logarithm).
```

Comment:
When I read $\log(f(x))$, I think I would naturally *assume* that it is calculated as $y=f(x)$, $\log(y)$. Hence the "but".  That is the intent, or why it makes sense to me.

Consider adding suggestions rather than copyingthe whole thing. Or is it not letting you?

Reply:
Sorry. I’m on mobile and was having trouble formatting these as suggestions.